tests/integration/security/util/framework.go

```diff
@@ -109,14 +116,20 @@ func EchoConfig(name string, ns namespace.Instance, headless bool, annos echo.An
 				Port:     8088,
 				Protocol: protocol.TCP,
 			},
+			{
+				Port:     8089,
+				Protocol: protocol.HTTPS,
+			},
 		},
 		Cluster: cluster,
 	}
 
 	// for headless service with selector, the port and target port must be equal
 	// Ref: https://kubernetes.io/docs/concepts/services-networking/service/#headless-services
 	if headless {
-		out.Ports[0].ServicePort = 8090
+		for i := range out.Ports {
```

Comment:
why need to change headless config?

Reply:
headless Service cannot have service port != target port, since there is no VIP/iptables rules to do the translation. This is a k8s thing, not istio

Reply:
and I made it do it for all ports since I added another port (https) that needs this logic